File: 1773-count-items-matching-a-rule/1773-count-items-matching-a-rule.php
class Solution {

    /**
     * @param String[][] $items
     * @param String $ruleKey
     * @param String $ruleValue
     * @return Integer
     */
    function countMatches($items, $ruleKey, $ruleValue) {
        $itemsMatchingRuleCount = 0;

        foreach($items as $item)
        {
            $matchesRuleForType = $ruleKey == 'type' && $item[0] == $ruleValue;
            $matchesRuleForColor = $ruleKey == 'color' && $item[1] == $ruleValue;
            $matchesRuleForName = $ruleKey == 'name' && $item[2] == $ruleValue;

            $matchesRule = $matchesRuleForType || $matchesRuleForColor || $matchesRuleForName;
            if($matchesRule)
                $itemsMatchingRuleCount++;
        }

        return $itemsMatchingRuleCount;
    }
}

File: 1773-count-items-matching-a-rule/1773-count-items-matching-a-rule.cs
public class Solution {
    public int CountMatches(IList<IList<string>> items, string ruleKey, string ruleValue) {
        Dictionary<string, int> mapRuleKeyToIndex = new () {
            { "type", 0 },
            { "color", 1 },
            { "name", 2 }
        };

        return items.Count(item => 
        {
            int index = mapRuleKeyToIndex[ruleKey];
            return item[index] == ruleValue;
        });
    }
}